ibazel/ibazel.go

```diff
@@ -448,9 +449,45 @@ func (i *IBazel) getInfo() (*map[string]string, error) {
 	return &res, nil
 }
 
+func (i *IBazel) realLocalRepositoryPaths() (map[string]string, error) {
+	info, _ := i.getInfo()
+	outputBase := (*info)["output_base"]
+	installBase := (*info)["install_base"]
+	externalPath := filepath.Join(outputBase, "external")
+
+	files, err := ioutil.ReadDir(externalPath)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error finding remote repositories directory: %v\n", err)
+		return map[string]string{}, err
+	}
+
+	localRepositories := map[string]string{}
+
+	for _, f := range files {
+		if !f.IsDir() && (f.Mode()&os.ModeSymlink) == os.ModeSymlink {
```

Comment:
I think we should probably break this section out into a per-os function in its own file and in the windows version print out something on the command line to inform the user that we don't support watching `local_repository` in Windows. WDYT?

Reply:
I've added a windows check but it will just silently ignore local_repositories, can you clarify how this might look if we break it out? This method is all intertwined right now.

If we are going to log something, it's not obvious to me how we detect this situation as I don't know how bazel deals with local repository rules on windows, therefore not sure how to detect they exist (our check for unix is basically "is it a symlink"). If I knew how to detect if they existed, we could probably just make windows work!

Reply:
> I've added a windows check but it will just silently ignore local_repositories, can you clarify how this might look if we break it out? This method is all intertwined right now.

Absolutely! The way I would go about this would be to create two new files in this directory, `ibazel_unix.go` and `ibazel_windows.go` like we did with [`main_unix.go`](https://github.com/bazelbuild/bazel-watcher/blob/master/ibazel/main_unix.go#L15) and [`main_windows.go`](https://github.com/bazelbuild/bazel-watcher/blob/master/ibazel/main_windows.go#L15). Inside those files you'll need 3 things:

* [ ] License header
* [ ] `// +build !windows` in the `ibazel_unix.go` file
* [ ] An implementation of a function with the same signature in both files.

As a total standin, you could just pass all the contextual variables you use and not return anything.

In the `_unix.go` file, you could have basically the same code:
```go
package ibazel

import (
 // stuff
)
func (i *IBazel) realLocalRepositoryPaths() (map[string]string, error) {
  // Put the entire implementation of this method in here 
}
```

and in `ibazel_windows.go` you could have:
```go
package ibazel

import (
  "github.com/bazelbuild/bazel-watcher/ibazel/log"
)

var alreadyNotifiedOfLocalRepositories bool
func (i *IBazel) realLocalRepositoryPaths() (map[string]string, error) {
  if !alreadyNotifiedOfLocalRepositories {
    // Put the entire implementation of this method in here 
    log.Banner(
      "iBazel does not support watching local_repository or --override_repository",
      "If this is a feature you'd like to add support for, please visit",
      "https://github.com/bazelbuild/bazel-watcher/issues/274"
    )
  }
  alreadyNotifiedOfLocalRepositories = true
}
```

> If we are going to log something, it's not obvious to me how we detect this situation as I don't know how bazel deals with local repository rules on windows, therefore not sure how to detect they exist (our check for unix is basically "is it a symlink"). If I knew how to detect if they existed, we could probably just make windows work!

I think we sidestep the issue of getting it to work in Windows this way without degrading peoples experience too much. If someone who knows more about the internals of Bazel wants to come along and add Windows support then there is a convenient place for them to drop it in.

Reply:
Done! Just to be clear, this warning will now always show when running ibazel in windows, regardless of if using local repositories or that flag (maybe this was the intention, just wanted to confirm!).